.github/workflows/python-tests.yaml

```diff
@@ -249,6 +249,190 @@ jobs:
+  run-tests-pydantic-v2:
+    name: pydantic v2, python:${{ matrix.python-version }}, ${{ matrix.database }}, ${{ matrix.pytest-options }}
+    needs: isinternal
+    strategy:
+      matrix:
+        database:
+          - "postgres:14"
+          - "sqlite"
+        os:
+          - ${{ needs.isinternal.outputs.status == 'true' && github.event_name == 'pull_request' && 'oss-test-runner' || 'ubuntu-latest' }}
+        python-version:
+          - "3.8"
+          - "3.11"
+        pytest-options:
+          - "--exclude-services"
+          - "--only-services"
+
+        include:
+          # Include Docker image builds on the service test run, and disallow the test
+          # suite from building images automaticlly in fixtures
+          - pytest-options: "--only-services"
+            build-docker-images: true
+
+        exclude:
+          # Do not run service tests with postgres
+          - database: "postgres:14"
+            pytest-options: "--only-services"
+
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 75
+
+    steps:
+      - name: Display current test matrix
+        run: echo '${{ toJSON(matrix) }}'
+
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        if: ${{ matrix.build-docker-images }}
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+          cache-dependency-path: "requirements*.txt"
+
+      - name: Pin requirements to lower bounds
+        if: ${{ matrix.lower-bound-requirements }}
+        # Creates lower bound files then replaces the input files so we can do a normal install
+        run: |
+          ./scripts/generate-lower-bounds.py requirements.txt > requirements-lower.txt
+          ./scripts/generate-lower-bounds.py requirements-dev.txt > requirements-dev-lower.txt
+          mv requirements-lower.txt requirements.txt
+          mv requirements-dev-lower.txt requirements-dev.txt
+
+      - name: Get image tag
+        id: get_image_tag
+        if: ${{ matrix.build-docker-images }}
+        run: |
+          SHORT_SHA=$(git rev-parse --short=7 HEAD)
+          tmp="sha-$SHORT_SHA-python${{ matrix.python-version }}"
+          echo "image_tag=${tmp}" >> $GITHUB_OUTPUT
+
+      - name: Build test image
+        if: ${{ matrix.build-docker-images }}
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          # TODO: We do not need the UI in these tests and we may want to add a build-arg to disable building it
+          #       so that CI test runs are faster
+          build-args: |
+            PYTHON_VERSION=${{ matrix.python-version }}
+            PREFECT_EXTRAS=[dev]
+          tags: prefecthq/prefect-dev:${{ steps.get_image_tag.outputs.image_tag }}
```

Comment:
Do you think it would be a good idea to append `pydanticv2` here?

Reply:
Thinking through that....I believe these tags are used in the test suite, so I don't think we can change them.  For what it's worth, these images aren't the ones that ever get pushed anywhere, they're just built locally for the tests.

But you _did_ make me realize that we aren't force-installing pydantic inside the test docker images, gonna do that now.